packages/engine/src/scene/components/ShadowComponent.ts

```diff
@@ -50,10 +52,9 @@ export const ShadowComponent = defineComponent({
 })
 
 function ShadowReactor({ root }: EntityReactorProps) {
+  const shadowComponent = useOptionalComponent(root.entity, ShadowComponent)
```

Comment:
I switched to useOptionalComponent before the effect, is this best practice?

Reply:
perfect!